docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,608 @@
+  but not necessarily across the entire tedge namespace/hierarchy.
+* If all child devices **can not** guarantee that uniqueness across the entire hierarchy,
+  then they are expected to register with thin-edge to get that unique ID generated.
+* Given that unique ID, thin-edge can identify its entire parent lineage and descendent child devices.
+  Thin-edge internally maintains this mapping information as child devices register with it.
+* If all child devices **can** guarantee ID uniqueness across the entire hierarchy,
+  they can skip this registration step.
+* Services need not have unique IDs across the entire hierarchy.
+  They just need to have unique IDs under their parent's namespace.
+
+The proposal is as follows:
+
+Topics to have device id as the top level prefix with distinction on the target: device or service as in:
+
+```
+tedge/<device-id>/<target-type>/...
+```
+
+The following demonstrates the different topic prefix for each of the target types:
+
+|Target|Topic Prefix|
+|------|-------|
+|tedge device|`tedge/main/device/`|
+|service on tedge device|`tedge/main/service/<service-id>/`|
+|child device|`tedge/<child-id>/device/`|
+|service on child device|`tedge/<child-id>/service/<service-id>/`|
+
+Where `main` is used as an alias for the tedge device id.
+
+**Why have the `/device/` subtopic level?**
+
+To make a clear distinction between `device` data and `service` data.
+This distinction makes it easier to subscribe to all device only data excluding the service data
+with simple subscription filters like `tedge/main/device/#`.
+Without that device suffix, something like `tedge/main/#` would include the services' data as well.
+
+
+**Why have the `/service/` subtopic level and not have the <service-id> directly?**
+
+Primarily for future-proofing as `service` is a new kind of child abstraction that we've added now.
+If another abstraction is introduced in the future, say `plugin`, they can be namespaced under a `plugin/` subtopic.
+It also makes the distinction from `device` data clearer.
+
+### Telemetry
+
+For telemetry data, the topics would be grouped under a `telemetry/` sub-topic as follows:
+
+|Type|Example|
+|------|-------|
+|Measurements|`<target_prefix>/telemetry/measurements`|
+|Events|`<target_prefix>/telemetry/events/<event-type>`|
+|Alarms|`<target_prefix>/telemetry/alarms/<alarm-type>/<severity>`|
+
+**Examples**
+
+The following shows an example of the measurement telemetry topics for each of the target types:
+
+|Target|Topic structure|
+|------|-------|
+|tedge device|`tedge/main/device/telemetry/measurements`|
+|service on tedge device|`tedge/main/service/<service-id>/telemetry/measurements`|
+|child device|`tedge/<child-id>/device/telemetry/measurements`|
+|service on child device|`tedge/<child-id>/service/<service-id>/telemetry/measurements`|
+
+**Why have the redundant `/telemetry` subtopic level?**
+
+To have a clear separation from commands or other kinds of data that might get added in future.
```

Comment:
Not really convinced that this is useful.

Reply:
The separation is most useful for writing ACL rules, where having a simple subscription mechanism can reduce human error when creating the ACL rules. Only specific clients/components should be given access to the `/commands` topics as you can control some device related functionality using a privileged user (e.g. privilege escalation by allowing a non-root user to restart the device)